src/oss-ci-cd-tooling/orb.yml

```diff
@@ -549,10 +549,29 @@ jobs:
             # Load our variables what we set earlier (e.g. DIST_TAG).
             source ~/scripts/apollo-setup-publish-env
 
-            echo "Publishing to npm under the $DIST_TAG tag."
+            # Get a list of packages that lerna manages
+            PACKAGES=`npx lerna list --json |jq -r '.[] | .name + "@" + .version'`
+
+            # Get all tags that we're trying to set. Lerna will be able to handle the first one,
+            # but if there are more than one, we'll need to handle it manually
+            # Note that we expect DIST_TAG to be comma delimited
+            DIST_TAG_ARR=($(echo $DIST_TAG | tr "," " "))
+            FIRST_TAG=${DIST_TAG_ARR[0]}
+
+            echo "Publishing to npm under the $FIRST_TAG tag."
 
             # Do the real, actual publish to npm.
-            DEBUG=lerna npx lerna publish --dist-tag="$DIST_TAG" from-git --no-verify-access --yes
+            DEBUG=lerna npx lerna publish --dist-tag="$FIRST_TAG" from-git --no-verify-access --yes
+
+            # iterate over remaining packages/tags and manually call npm dist-tag for each of them
+            DIST_TAG_ARR=("${DIST_TAG_ARR[@]:1}")
+            for tag in ${DIST_TAG_ARR[@]}; do
+              for package in $PACKAGES; do 
```

Comment:
We could avoid both `jq` and the **`for package`** loop if we use Lerna's own logic to "run a command on each current version of a package" logic (offered in [`lerna exec`](https://github.com/lerna/lerna/blob/main/commands/exec/README.md)) after the initial `lerna publish` (keeping that first `publish` using the new `FIRST_TAG` variable you've defined) and using `lerna exec` to apply subsequent tags:

```
npx lerna exec --no-private --no-bail --concurrency 1 --stream -- echo npm dist-tag add "\$LERNA_PACKAGE_NAME" $TAG_FROM_LOOP
```

> You'll note that I included the `echo` command in front to simulate a "dry run" and that I used `TAG_FROM_LOOP` as a reference to your `tag` variable in the `for tag in` loop from your code.  You should be able to run it locally in the repo and it should work!

That would ensure that Lerna's configuration (including its "ignore", "private" package designators, registry overrides, etc.) continues to be reliably obeyed and wouldn't rely on us using their JSON representation to implement what I'd claim is the same behavior in our own ad-hoc way.

---

Breaking down the above command (I took this from my shell history as something I used to use specifically for exactly this) it's saying:

Run (`exec`) the command after `--` with a concurrency of 1 (meaning, don't try to fan out and run it on all packages at the same time) without bailing if any **one** command fails and exclude any package that is marked as private.

In theory, we could _probably_ run without the `--concurrency` flag and speed things up a bit, but I suggested using `1` since we'd probably want to avoid stampeding things if there is a failure condition on even just one.  Concurrency above `1` would just makes doing an analysis via logs more challenging. 😉 

Thoughts?